src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessService.java

```diff
@@ -65,9 +80,16 @@ public AccessLevel getAccessLevel(UserInfo userInfo) {
 
     public Optional<List<Long>> getGrantedCaseIdsForRestrictedRoles() {
         if (userCanOnlyAccessExplicitlyGrantedCases()) {
-            return Optional.of(caseUserRepository.findCasesUserIdHasAccessTo(userRepository.getUserId()));
+            final List<Long> caseIds;
+            if (applicationParams.getEnableAttributeBasedAccessControl()) {
+                caseIds = roleAssignmentService.getCaseIdsForAGivenUser(userRepository.getUserId())
```

Comment:
I think this needs additional checks defined in the https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1460559903#AccessControlScopeofDelivery-AggregatedSimpleCaseRoleClause Please have a look. You need to create a method to get role assignments and filter them to have a collection of "A 'simple' case role assignments". getCaseIdsForAGivenUser(String userId) only filters on roleType=CASE and timestamp.